warehouse/query-core/src/main/java/datawave/core/iterators/DatawaveFieldIndexRangeIteratorJexl.java

```diff
@@ -109,60 +117,86 @@ public String toString() {
-        startKey = new Key(rowId, fiName, new Text(boundingFiRangeStringBuilder.toString()));
-        
+        return new Key(rowId, fiName, new Text(boundingFiRangeStringBuilder.toString()));
+    }
+    
+    private Key createUpperBoundKey(Text rowId, Text fiName, Text upperBound, boolean isUpperInclusive) {
         // we cannot simply use endKeyInclusive in the Range as the datatype and UID follow the value in the keys
         // hence we need to compute the max possibly value that would be inclusive
         this.boundingFiRangeStringBuilder.setLength(0);
-        if (upperInclusive) {
+        if (isUpperInclusive) {
             this.boundingFiRangeStringBuilder.append(upperBound).append(ONE_BYTE);
         } else {
             String upperString = upperBound.toString();
-            // in case of composite ranges, use the discrete index type to get the inclusive bound
-            if (compositeSeeker != null && compositeSeeker.getFieldToDiscreteIndexType().get(fiName.toString()) != null) {
-                DiscreteIndexType discreteIndexType = compositeSeeker.getFieldToDiscreteIndexType().get(fiName.toString());
-                this.boundingFiRangeStringBuilder.append(discreteIndexType.decrementIndex(upperString));
-            } else {
-                this.boundingFiRangeStringBuilder.append(upperString.substring(0, upperString.length() - 1));
-                this.boundingFiRangeStringBuilder.append((char) (upperString.charAt(upperString.length() - 1) - 1));
-            }
+            this.boundingFiRangeStringBuilder.append(upperString.substring(0, upperString.length() - 1));
+            this.boundingFiRangeStringBuilder.append((char) (upperString.charAt(upperString.length() - 1) - 1));
             this.boundingFiRangeStringBuilder.append(Constants.MAX_UNICODE_STRING);
         }
-        endKey = new Key(rowId, fiName, new Text(boundingFiRangeStringBuilder.toString()));
-        if (isNegated()) {
-            Key startFi = new Key(rowId, fiName);
-            Key endFi = new Key(rowId, new Text(fiName.toString() + '\0'));
-            List<Range> rangeList = new ArrayList<>(new RangeSplitter(new Range(startFi, true, startKey, true), getMaxRangeSplit() / 2));
-            rangeList.addAll(new RangeSplitter(new Range(endKey, true, endFi, true), getMaxRangeSplit() / 2));
-            return rangeList;
-        } else {
-            return new RangeSplitter(new Range(startKey, true, endKey, true), getMaxRangeSplit());
-        }
+        return new Key(rowId, fiName, new Text(boundingFiRangeStringBuilder.toString()));
     }
     
     protected Range buildCompositeSafeFiRange(Text rowId, Text fiName, Text fieldValue) {
-        Key startKey = new Key(rowId, fiName, new Text(fieldValue));
-        Key endKey = new Key(rowId, fiName, new Text(upperBound));
-        return new Range(startKey, lowerInclusive, endKey, upperInclusive);
+        if (subRanges != null && !subRanges.isEmpty()) {
```

Comment:
Yeah, it's needed when we do composite seeking.  We need to know the original range that was used so that we can accurately increment the bounds of the ranges as we skip along.